File: 717-1-bit-and-2-bit-characters/1-bit-and-2-bit-characters.py
class Solution:
    def isOneBitCharacter(self, bits: List[int]) -> bool:
        n = len(bits)

        i = 0
        while i < n:
            if 1 == bits[i]:
                i += 1
            elif i == n - 1:
                return True
            i += 1

        return False

File: 717-1-bit-and-2-bit-characters/1-bit-and-2-bit-characters.kt
class Solution {
    fun isOneBitCharacter(bits: IntArray): Boolean {
        val n: Int = bits.size

        var i: Int = 0
        while (i < n) {
            if (1 == bits[i]) {
                i++
            } else if (i == n - 1) {
                return true
            }
            i++
        }

        return false
    }
}